pkg/cluster/cluster.go

```diff
@@ -295,6 +295,8 @@ func (c *Cluster) run(stopC <-chan struct{}) {
 				break
 			}
 
+			c.gcServices()
```

Comment:
can we make this a monitor routine? instead of running it for every reconciliation?